src/include/txn.i

```diff
@@ -963,7 +970,7 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
     }
 
     /* Store the stop time pair of the history store record that is returning. */
-    if (WT_TIME_WINDOW_HAS_STOP(&tw) && WT_IS_HS(S2BT(session))) {
+    if (!have_stop_tw && WT_TIME_WINDOW_HAS_STOP(&tw) && WT_IS_HS(S2BT(session))) {
```

Comment:
The only thing I'm confused about is the changes in this file.

Is it expected to have `have_stop_tw && WT_TIME_WINDOW_HAS_STOP(&tw)`? It seems like that implies double tombstone.

Reply:
We place globally visible tombstones on top the pre-existing history store tombstone.